app/lib/screens/app_onboarding/preview.dart

```diff
@@ -66,15 +66,17 @@ class Preview {
     // delete study subscription and progress
     if (containsQueryPair('cmd', 'reset')) {
       if (selectedStudyObjectId != null) {
-        final StudySubject subject = await SupabaseQuery.getById<StudySubject>(
-          selectedStudyObjectId!,
-          selectedColumns: [
-            '*',
-            'study!study_subject_studyId_fkey(*)',
-            'subject_progress(*)',
-          ],
-        );
-        subject.delete();
+        final List<StudySubject> subjects =
+            await SupabaseQuery.getAll<StudySubject>();
+
+        final List<StudySubject> subjectsToDelete = subjects
+            .where((subject) =>
+                subject.userId == Supabase.instance.client.auth.currentUser!.id)
+            .toList();
+
+        subjectsToDelete.forEach((subject) async {
+          await subject.delete();
+        });
```

Comment:
Is this related to this PR? I remember we wanted to clear all study_subjects for the user who is logged into the Designer, when they use the Reset button in the preview screen. This seems to fail for me:
`
[PreviewApp]: Failed fetching subject: PostgrestException(message: invalid input syntax for type uuid: "", code: 22P02, details: Bad Request, hint: null)`

Reply:
Ah I see this is probably part of https://github.com/hpi-studyu/studyu/pull/699